### PR TITLE
check also the Authorization for content

### LINKED ISSUE #14: EasyAuth doesn't authenticate bearer tokens issued for applications
Any token coming from user login seems to work but tokens for applications which use client credentials won't.

### PR DESCRIPTION
**Pull request by paule96**
this try to fix #14 .

Hi @Suzuko123,

I hope this will fix your problem, but I was not able to test this. Maybe you can try it and give us feedback 🤞

### CODE REVIEW

**Comment by paule96** [comment]
so I have work hard to get this work in progress but it's working.

The only thing to do is the enablement of the custom options. That is currently not working.

@Suzuko123 this should work for you if you don't need a custom path to the local authme json file.

**Comment by paule96** [comment]
This PR have some breaking changes so we should change the major version

**Comment by paule96** [comment]
So know we are close to done. I will discuss maybe later this weak with @kirkone some naming changes.

The biggest renaming of the current development is from `service` to provider. 

There is also one point to discuss. If we want the `EasyAuthWithAuthMeService` to get a normal provider like each other provider. so you can disable it and can't force it as the last provider.

**Comment by paule96** [comment]
This PR now also solves the issue #3 .

We must discuss how open our API should be. Because currently the most things are public. That mean we can't change it without bumbing the Major version...

File: README.md [hunk 1]
```diff
@@ -83,23 +83,60 @@ public string UserName()
 
 If you want to add roles to the `User` property you can have a look in `Transformers/ClaimsTransformer.cs` in the Sample project. There you can see an example how to get started with this.
 
-### Custom options
+### Configure options via configuration (recommended)
+
+You can use the default behavior of asp.net core to configure EasyAuth. You must only change in your `Startup.cs` the `.AddEasyAuth()` to `.AssEasyAuth(this.Configuration)`.
```
Comment (kirkone) [inline README.md]: This should be `Add` not `Ass`

File: README.md [hunk 2]
```diff
@@ -83,23 +83,60 @@ public string UserName()
...
+        "Enabled": true
+      }
+    ]
+  }
+```
+
+Here are some notes to the json above:
+
+- each provider is disabled in default so you must enabled it
+- you can create own providers but this must implement `IEasyAuthAuthentificationService`. But you must also activate them here. (Don't but them in the DI. This package will do this by it's own.)
```
Comment (kirkone) [inline README.md]: "Don't but them in the DI" -> "Don't put them in the DI"

File: README.md [hunk 3]
```diff
@@ -83,23 +83,60 @@ public string UserName()
...
 
 You can provide additional options vor the middleware:
 
 ```
 ).AddEasyAuth(
                 options =>
                 {
-                    // Override the default claim for the User.Identity.Name field 
-                    options.NameClaimType = ClaimTypes.Email;
+                    // Override the auth endpoint
```
Comment (kirkone) [inline README.md]: please fix indent on this block

File: src/KK.AspNetCore.EasyAuthAuthentication.Sample/Transformers/ClaimsTransformer.cs [hunk 4]
```diff
@@ -19,7 +19,7 @@ public class ClaimsTransformer : IClaimsTransformation
         public ClaimsTransformer(
             IRepository repository,
             IHttpContextAccessor httpContextAccessor,
-            IMemoryCache cache
+            IMemoryCache cache
```
Comment (kirkone) [inline src/KK.AspNetCore.EasyAuthAuthentication.Sample/Transformers/ClaimsTransformer.cs]: Please fix whitespaces

File: src/KK.AspNetCore.EasyAuthAuthentication.Sample/Transformers/ClaimsTransformer.cs [hunk 5]
```diff
@@ -31,11 +31,11 @@ public async Task<ClaimsPrincipal> TransformAsync(ClaimsPrincipal principal)
         {
             if (principal.Identity.IsAuthenticated)
             {
-                var claimsIdentity = (ClaimsIdentity)principal.Identity;
+                var claimsIdentity = (ClaimsIdentity)principal.Identity;
```
Comment (kirkone) [inline src/KK.AspNetCore.EasyAuthAuthentication.Sample/Transformers/ClaimsTransformer.cs]: Please fix whitespaces

**Comment by paule96** [comment]
change target branch to `2.0.0-beta`

**Comment by paule96** [comment]
hi @kirkone,

I have update to code now finally so we should merge this as soon as possible to the 2.0.0 branch so we can release a preview package.

Important I have change the build yml. That's very important!

File: GlobalSuppressions.cs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+// This file is used by Code Analysis to maintain SuppressMessage
+// attributes that are applied to this project.
+// Project-level suppressions either have no target or are given
+// a specific target and scoped to a namespace, type, member, etc.
+
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0022:Use expression body for methods", Justification = "It's for better understanding and for simpler debugging")]
```
Comment (kirkone) [inline GlobalSuppressions.cs]: Why is this necessary?

Comment (paule96) [reply GlobalSuppressions.cs]: because now we don't get warnings for methods like this:

```csharp
public void blup(){
   Console.WriteLine("hello world");
}
```

Because this can be converted to:

```csharp
public void blup => Console.WriteLine("hello world");
```

But I hat this because you can't debug this single line good. And it is harder to read. explicit in long method calls.

Comment (kirkone) [reply GlobalSuppressions.cs]: I will update the EditorConfig later.
that should solve this problem
